addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveClockCommandClass.java

```diff
@@ -64,10 +134,70 @@ public void handleApplicationCommandRequest(SerialMessage serialMessage, int off
         logger.debug("NODE {}: Received clock command (v{})", this.getNode().getNodeId(), this.getVersion());
         int command = serialMessage.getMessagePayloadByte(offset);
         switch (command) {
+            case CLOCK_REPORT:
+                int day = serialMessage.getMessagePayloadByte(offset + 1) >> 5;
+                int hour = serialMessage.getMessagePayloadByte(offset + 1) & 0x1f;
+                int minute = serialMessage.getMessagePayloadByte(offset + 2);
+                String days[] = new DateFormatSymbols().getWeekdays();
+                int javaDay = day == 7 ? 1 : day + 1;
+                logger.debug(String.format("NODE %d: Received clock report: %s %02d:%02d", getNode().getNodeId(),
+                        days[javaDay], hour, minute));
+
+                Calendar cal = Calendar.getInstance();
+                cal.set(Calendar.DAY_OF_WEEK, javaDay);
+                cal.set(Calendar.HOUR_OF_DAY, hour);
+                cal.set(Calendar.MINUTE, minute);
+                Date nodeTime = cal.getTime();
+                long clockOffset = (nodeTime.getTime() - System.currentTimeMillis()) / 1000 / 60;
```

Comment:
I wonder if this is the best approach for the protocol layers to make this decision. I think it's better to simply report the time and let the application layers calculate the offset and decide what to do about it.